core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2345,17 +2345,19 @@ private[spark] class RedirectThread(
  */
 private[spark] class CircularBuffer(sizeInBytes: Int = 10240) extends java.io.OutputStream {
   var pos: Int = 0
+  var isBufferFull = false
   var buffer = new Array[Int](sizeInBytes)
```

Comment:
I think we have more problems with this class. This should be `val buffer = new Array[Byte](sizeInBytes)`. It's 4x larger than it needs to be.